docs/guide/tutorials/create-indexes.txt

```diff
@@ -1,3 +1,241 @@
+
+.. code-block:: js
+
+   async function createTextIndex(db) {
+     // Get the documents collection
+     const collection = db.collection('users');
+
+     // Create the index
+     const result = await collection.createIndex({ comments : "text" });
+     console.log(result);
+     return result;
+   );
+
+Create a Hashed Index
+---------------------
+
+To specify a :manual:`hashed </core/index-hashed/>` index key,
+use the ``hashed`` method.
+
+This example specifies a hashed index key for the ``timestamp`` field:
+
+.. code-block:: js
+
+   async function createHashedIndex(db) {
+     // Get the documents collection
+     const collection = db.collection('users');
+
+     // Create the index
+     const result = await collection.createIndex({ timestamp : "hashed" });
+     console.log(result);
+     return result;
+   };
+
+Create Geospatial Indexes
+-------------------------
+
+There are also helpers for creating the index keys for the various
+geospatial indexes supported by mongodb.
+
+Create a ``2dsphere`` Index
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To specify a :manual:`2dsphere </core/2dsphere/>`
+index key, use one of the ``geo2dsphere`` methods.
+
+This example specifies a 2dsphere index on the ``location`` field:
+
+.. code-block:: js
+
+   async function create2dSphereIndex(db) {
+     // Get the documents collection
+     const collection = db.collection('users');
+
+     // Create the index
+     const result = await collection.createIndex({ location : "2dsphere" });
+     console.log(result);
+     return result;
+   };
+
+Create a ``2d`` Index
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To specify a :manual:`2d </core/2d/>` index key, use the ``geo2d``
+method.
+
+.. important::
+
+   A 2d index is for data stored as points on a two-dimensional plane
+   and is intended for legacy coordinate pairs used in MongoDB 2.2 and
+   earlier.
+
+
+This example specifies a 2d index on the ``points`` field:
+
+.. code-block:: js
+
+   async function create2dIndex(db) {
+     // Get the documents collection
+     const collection = db.collection('users');
+
+     // Create the index
+     const result = await collection.createIndex({ points : "2d" });
+     console.log(result);
+     return result;
+   };
+
+IndexOptions
+------------
+
+In addition to the index specification document, ``createIndex``
+method can take an index options document, such as to create unique
+indexes or partial indexes.
+
+Create a Unique Index
+^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: js
+
+   async function createUniqueIndex(db) {
+     // Get the documents collection
+     const collection = db.collection('users');
+
+     // Create the index
+     const result = await collection.createIndex(
+       { lastName : -1, dateOfBirth : 1 },
+       { unique:true }
+     );
+     console.log(result);
+     return result;
+   };
+
+Create a Partial Index
+^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: js
+
+   async function createPartialIndex(db, callback) {
```

Comment:
the `callback` in `createPartialIndex(db, callback)` isn't used. Should it be used or removed?